pkg/storage/mysql/mysql.go

```diff
@@ -391,16 +391,41 @@ func (m *MySQL) ReadAuthorizationModels(ctx context.Context, store string, opts
 		}
 	}
 
-	// TODO: make this concurrent with a maximum of 5 goroutines. This may be helpful:
-	// https://stackoverflow.com/questions/25306073/always-have-x-number-of-goroutines-running-at-any-time
+	// Initialize channels and worker pool
+	modelIDChan := make(chan string, numModelIDs)
+	modelsChan := make(chan *openfgav1.AuthorizationModel, numModelIDs)
+	errChan := make(chan error, 1)
+
+	// Start a fixed number of goroutines
+	numWorkers := 5
+	for i := 0; i < numWorkers; i++ {
+		go func() {
+			for id := range modelIDChan {
+				model, err := m.ReadAuthorizationModel(ctx, store, id)
+				if err != nil {
+					errChan <- err
+					return
+				}
+				modelsChan <- model
+			}
+		}()
+	}
+
+	// Send model IDs to the workers
+	for _, id := range modelIDs[:numModelIDs] {
+		modelIDChan <- id
+	}
+	close(modelIDChan)
+
+	// Collect results from the workers
 	models := make([]*openfgav1.AuthorizationModel, 0, numModelIDs)
-	// We use numModelIDs here to avoid retrieving possibly one extra model.
 	for i := 0; i < numModelIDs; i++ {
-		model, err := m.ReadAuthorizationModel(ctx, store, modelIDs[i])
-		if err != nil {
+		select {
+		case model := <-modelsChan:
+			models = append(models, model)
```

Comment:
Compared to `main`, does this implementation change the order of the models in the response? I'd assume that in `main` responses are sorted by model_id whereas here they are not.

We should try to keep the response sorted by descending ID. (Personally, I don't think it matters, but we may have customers that rely on the order)

1. Add a test here https://github.com/openfga/openfga/blob/164963facc61241e1378d8fefb6632c700407163/pkg/storage/test/storage.go#L26-L49 
2. Document here https://github.com/openfga/openfga/blob/78422af76964abbfe524002a7dbf1f01e8ec8979/pkg/storage/storage.go#L140-L141

If it's not possible we can always document in the release notes that this is breaking change

Reply:
Regarding sorting, I think we would need something to check about the concurrency sync is completed or not (i.e, all channels are done or not). For that, we can use `sync.waitGroup` (but it would require a separate import). Or we could the old-school way (like following), then sort it by descending order.

```go
// Initialize channels
    modelIDChan := make(chan string, numModelIDs)
    doneChan := make(chan bool, numModelIDs)
    models := make([]*openfgav1.AuthorizationModel, numModelIDs)
    errChan := make(chan error, 1)

    // Start a fixed number of worker goroutines
    numWorkers := 5
    for i := 0; i < numWorkers; i++ {
        go func() {
            for id := range modelIDChan {
                model, err := m.ReadAuthorizationModel(ctx, store, id)
                if err != nil {
                    errChan <- err
                    return
                }
                for index, modelID := range modelIDs[:numModelIDs] {
                    if modelID == id {
                        models[index] = model
                        break
                    }
                }
                doneChan <- true
            }
        }()
    }

    // Send model IDs to the workers
    for _, id := range modelIDs[:numModelIDs] {
        modelIDChan <- id
    }
    close(modelIDChan)

    // Wait for all goroutines to finish
    for i := 0; i < numModelIDs; i++ {
        select {
        case <-doneChan:
            // One goroutine finished
        case err := <-errChan:
            // Return early on error
            return nil, nil, err
        }
    }

    // Sort models in descending order by their ID
    sort.Slice(models, func(i, j int) bool {
        return models[i].GetId() > models[j].GetId()
    })

    return models, token, nil
```


wdyt @miparnisari ?